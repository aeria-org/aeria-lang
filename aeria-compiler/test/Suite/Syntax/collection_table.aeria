collection User {
  table {
    name
    email
  }

  properties {
    name      str
    email     str
    password  str
  }
}
