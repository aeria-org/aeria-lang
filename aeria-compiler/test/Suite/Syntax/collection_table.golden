{
  "kind": "Program",
  "collections": [
    {
      "table": [
        {
          "propertyName": {
            "value": "name",
            "kind": "PropertyName"
          },
          "kind": "TableItem"
        },
        {
          "propertyName": {
            "value": "email",
            "kind": "PropertyName"
          },
          "kind": "TableItem"
        }
      ],
      "properties": [
        {
          "type": {
            "kind": "PString"
          },
          "name": {
            "value": "name",
            "kind": "PropertyName"
          },
          "kind": "Property"
        },
        {
          "type": {
            "kind": "PString"
          },
          "name": {
            "value": "email",
            "kind": "PropertyName"
          },
          "kind": "Property"
        },
        {
          "type": {
            "kind": "PString"
          },
          "name": {
            "value": "password",
            "kind": "PropertyName"
          },
          "kind": "Property"
        }
      ],
      "name": {
        "value": "user",
        "kind": "CollectionName"
      },
      "kind": "Collection"
    }
  ]
}
